### PR TITLE
Fixing an issue with `Highlight`

### LINKED ISSUE #6643: `Highlight` returns nothing when `query` is an empty string
### Description

When I use `Highlight` with an empty query, I expect the text to be returned without a highlighted effect

### Link to Reproduction

https://codesandbox.io/s/twilight-monad-7pk5qj?file=/src/index.tsx

### Steps to reproduce

1. Create a `Highlight` component with an empty string for the `query` prop

### Chakra UI Version

2.2.9

### Browser

- `*`

### Operating System

- [X] macOS
- [X] Windows
- [ ] Linux

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by SpencerKaiser**
Closes #6643 

## 📝 Description
Fixes an issue where an empty query string would prevent `Highlight` from returning anything

## ⛳️ Current behavior (updates)
The component returns nothing

## 🚀 New behavior
The hook that `Highlight` is reliant on will ALWAYS return something, but when an empty string is provided the entire text is **not** a match for the query

## 💣 Is this a breaking change (Yes/No):
No

### CODE REVIEW

File: packages/components/layout/src/highlight.tsx [hunk 1]
```diff
@@ -36,7 +36,7 @@ function buildRegex(query: string[]) {
 function highlightWords({ text, query }: Options): Chunk[] {
   const regex = buildRegex(Array.isArray(query) ? query : [query])
   if (!regex) {
-    return []
+    return [{ text: text, match: false}]
```
Comment (SpencerKaiser) [inline packages/components/layout/src/highlight.tsx:39]:
```suggestion
    return [{ text, match: false}]
```

**Comment by segunadebayo** [comment]
Hi @SpencerKaiser. Kindly update the tests snapshots as well

**Comment by SpencerKaiser** [comment]
@segunadebayo done! Lemme know if there's anything else you need!

Sorry for missing the broken tests; I was kinda flying blind because I haven't used `pnpm` before and [the docs were missing something fairly important](https://github.com/chakra-ui/chakra-ui/pull/6650) 😉

**Comment by SpencerKaiser** [comment]
@segunadebayo thanks for merging! When will this be included in a release? I tried figuring out the release flow for this repo but didn't have much luck 😬

**Pull request merged**